File: archive/m/mojiscript/baklava.mjs
import log from 'mojiscript/console/log'
import pipe from 'mojiscript/core/pipe'
import run from 'mojiscript/core/run'
import map from 'mojiscript/list/map'
import range from 'mojiscript/list/range'

const numSpaces = n => Math.abs(n)
const numStars = n => 21 - 2 * numSpaces (n)
const baklavaLine = n => ' '.repeat(numSpaces (n)) + '*'.repeat(numStars (n))

const main = pipe([
    range (-10) (11),
    map (baklavaLine),
    map (log)
])

run({main})

File: archive/e/eta/Baklava.hs
module Main where

baklavaLine :: Int -> String
baklavaLine n = (replicate numSpaces ' ') ++ (replicate numStars '*') ++ "\n"
    where
        numSpaces = abs(n - 10)
        numStars = 21 - 2 * numSpaces

baklava :: String -> Int -> String
baklava s 0 = s ++ baklavaLine 0
baklava s n = s ++ baklavaLine n ++ baklava s (n - 1)

main :: IO ()
main = putStr (baklava "" 20)
